lume_model/models/gp_model.py

```diff
@@ -0,0 +1,412 @@
+import os
+import logging
+import warnings
+
+from pydantic import field_validator
+
+import torch
+from torch.distributions import Distribution as TDistribution
+from botorch.models import SingleTaskGP, MultiTaskGP, ModelListGP
+from gpytorch.mlls import ExactMarginalLogLikelihood
+from gpytorch.distributions import MultivariateNormal as GPMultivariateNormal
+from botorch.models.transforms.input import ReversibleInputTransform
+from botorch.models.transforms.outcome import OutcomeTransform
+from linear_operator.utils.cholesky import psd_safe_cholesky
+from linear_operator.operators import DiagLinearOperator
+
+from lume_model.models.prob_model_base import (
+    ProbModelBaseModel,
+    TorchDistributionWrapper,
+)
+
+
+logger = logging.getLogger(__name__)
+
+
+class GPModel(ProbModelBaseModel):
+    """
+    LUME-model class for GP models.
+    This supports Botorch's SingleTask, MultiTask, and ModelListGP models.
+
+    Note that if the Botorch model was trained with input_transform or outcome_transform set to anything other than None,
+    those transformations will be automatically applied, as well as any input_transformers or output_transformers
+    passed to this class. For ModelListGP, the passed input_transformers and output_transformers will be applied to all
+    models in the list. If different transformers are needed for different models, the models should be instantiated
+    separately using Botorch's input_transform and outcome_transform attributes before creating the model list.
+
+    Args:
+        model: A single task GPyTorch model or BoTorch model.
+        input_transformers: List of input transformers to apply to the input data. Optional, default is None.
```

Comment:
should specify here that passing a list of transformers here applies them sequentially to the inputs / outputs